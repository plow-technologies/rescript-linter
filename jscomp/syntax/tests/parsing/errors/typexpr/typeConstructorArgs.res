type node<'a> = {
  _value: Js.Nullable.value('a)
}

type t<'a> = Belt.Map.t('a)
type t<'a> = private Belt.Map.t('a)

type t = option<<node<int>>
type t = option(<node<int>>)
