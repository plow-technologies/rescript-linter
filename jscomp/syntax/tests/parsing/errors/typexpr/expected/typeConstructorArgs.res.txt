
  Syntax error!
  tests/parsing/errors/typexpr/typeConstructorArgs.res:2:28-31

  1 │ type node<'a> = {
  2 │   _value: Js.Nullable.value('a)
  3 │ }
  4 │ 

  Type parameters require angle brackets:
  Js.Nullable.value<'a>


  Syntax error!
  tests/parsing/errors/typexpr/typeConstructorArgs.res:5:24-27

  3 │ }
  4 │ 
  5 │ type t<'a> = Belt.Map.t('a)
  6 │ type t<'a> = private Belt.Map.t('a)
  7 │ 

  Type parameters require angle brackets:
  Belt.Map.t<'a>


  Syntax error!
  tests/parsing/errors/typexpr/typeConstructorArgs.res:6:32-35

  4 │ 
  5 │ type t<'a> = Belt.Map.t('a)
  6 │ type t<'a> = private Belt.Map.t('a)
  7 │ 
  8 │ type t = option<<node<int>>

  Type parameters require angle brackets:
  Belt.Map.t<'a>


  Syntax error!
  tests/parsing/errors/typexpr/typeConstructorArgs.res:9:28

   7 │ 
   8 │ type t = option<<node<int>>
   9 │ type t = option(<node<int>>)
  10 │ 

  I'm not sure what to parse here when looking at ")".

type nonrec 'a node = {
  _value: 'a Js.Nullable.value }
type nonrec 'a t = 'a Belt.Map.t
type nonrec 'a t = private 'a Belt.Map.t
type nonrec t = int node option
type nonrec t = int node option